src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -175,6 +216,8 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
       try {
         const workspace = await Ajax(signal).Workspaces.workspace(namespace, name).details()
         workspaceStore.set(workspace)
+        const res = await checkBucketAccess(signal, namespace, name)
+        setHasBucketAccess(res)
```

Comment:
Because the rest of the workspace data gets cached in `workspaceStore`, this only gets run the first time you hit a workspace page, and skipped on subsequent pages. However, the wrapper gets unmounted, so this state is cleared. As a result, if you click away and then back to the dashboard, the bucket link will be disabled even though you should have access.

It might be better to merge this flag into the workspace object itself so it can share the same lifecycle as the rest of the data.